### PR TITLE
add ColorUtilityService

### LINKED ISSUE #525: create ColorService
- [x] all the global functions inside https://github.com/owncloud/calendar/blob/63ccc7b39753f8c0a4bf7fe0b009811e82917ca8/js/app/directives/colorpickerDirective.js should be moved into a ColorService
- [x] use 🇺🇸 spelling

### PR DESCRIPTION
**Pull request by georgehrke**
fixes https://github.com/owncloud/calendar/issues/525

### CODE REVIEW

File: js/app/service/colorutilityservice.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	/**
+	 * convert three Numbers to rgb hex string
+	 * @param r
+	 * @param g
+	 * @param b
+	 * @returns {string}
+	 */
+	this.rgbToHex = function(r, g, b) {
+		if(Array.isArray(r)) {
+			[r, g, b] = r;
```
Comment (georgehrke) [inline js/app/service/colorutilityservice.js]: - [ ] get rid of ES6 array restructuring

File: js/app/service/colorutilityservice.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	 * List of default colors
+	 * @type {string[]}
+	 */
+	this.colors = [];
+
+	if (typeof String.prototype.toHsl === 'function') {
+		//0 40 80 120 160 200 240 280 320
+		var hashValues = ['15', '9', '4', 'b', '6', '11', '74', 'f', '57'];
+		angular.forEach(hashValues, function(hashValue) {
+			var hslColor = hslToRgb(hashValue.toHsl());
```
Comment (georgehrke) [inline js/app/service/colorutilityservice.js]: - [ ] destructure array before handing over to hslToRgb()

File: js/public/app.js [hunk 3]
```diff
@@ -3929,6 +3929,115 @@ app.service('CalendarService', ['DavClient', 'Calendar', function(DavClient, Cal
...
+			parseInt(b, 10).toString(16);
+	};
+
+	/**
+	 * generates a random color
+	 * @returns {string}
+	 */
+	this.randomColor = function() {
+		if (typeof String.prototype.toHsl === 'function') {
+			return self.rgbToHex(hslToRgb(Math.random().toString().toHsl()));
```
Comment (georgehrke) [inline js/public/app.js]: - [ ] destructure array before handing over to hslToRgb()

**Comment by georgehrke** [comment]
- [x] add unit tests
- [x] update makefile to ignore certain hsl_rgb_converter files

**Comment by skjnldsv** [comment]
Seems great! :)

**Pull request merged**